Sources/MapboxDirections/RouteLeg.swift

```diff
@@ -141,6 +159,19 @@ open class RouteLeg: Codable {
         return segmentRangesByStep
     }()
     
+    /**
+     Segments for each Intersection along the route.
+     
+     Ordered by `steps`, inside one `step` - ordered by `Intersection`.  `nil` value means no index was provided. Index values correspond to `route`'s `shape` elements.
+     */
+    public private(set) lazy var intersectionsIndexesByStep: [[Int?]?] = {
```

Comment:
Consider renaming to `intersectionIndicesByStep`, though #478 proposes replacing this property with `RouteStep.segmentRangesByIntersection`, since it’s trivial to aggregate those step-based arrays at the leg level.